test/unit/compose/app.spec.ts

```diff
@@ -602,6 +602,65 @@ describe('compose/app', () => {
 			// The network should not be created again
 			expectNoStep('createNetwork', steps);
 		});
+
+		it('should create a config-only network if network_mode is host for all services', async () => {
+			const svcOne = await createService({
+				appId: 1,
+				serviceName: 'one',
+				composition: { network_mode: 'host' },
+			});
+			const svcTwo = await createService({
+				appId: 1,
+				serviceName: 'two',
+				composition: { network_mode: 'host' },
+			});
+			const current = createApp({
+				services: [svcOne, svcTwo],
+				networks: [],
+			});
+			const target = createApp({
+				services: [svcOne, svcTwo],
+				networks: [],
+				isTarget: true,
+			});
+
+			const steps = current.nextStepsForAppUpdate(defaultContext, target);
+
+			const [createNetworkStep] = expectSteps('createNetwork', steps);
+			expect(createNetworkStep)
+				.to.have.property('target')
+				.that.has.property('config')
+				.that.deep.includes({ configOnly: true });
+		});
+
+		it('should not create a config-only network if network_mode: host is not specified for any service', async () => {
```

Comment:
Maybe you could add a third test here for the "empty release" case, to ensure that a default network is created with `configOnly: true`?

Reply:
Good shout! Will do